src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -355,6 +369,10 @@ private void setAttributesIfNecessary(Object record, Message<?> message) {
 				messageBuilder.copyHeadersIfAbsent(messageToUse.getHeaders());
 			}
 
+			if (CheckpointMode.manual == KclMessageDrivenChannelAdapter.this.checkpointMode) {
+				messageBuilder.setHeader(AwsHeaders.CHECKPOINTER, checkpointer);
```

Comment:
I wonder if we can adapt a `RecordProcessorCheckpointer` to the existing `Checkpointer` abstraction in the project...

Reply:
Let's don't forget to add an issue for this!
I really would like to have an aligned model, when some one decides to switch from one Kinesis Channel Adapter to another.